File: utils/RedisUtil.java
package com.runtimebay.utils;

import org.yaml.snakeyaml.Yaml;
import redis.clients.jedis.Jedis;
import redis.clients.jedis.JedisPool;
import redis.clients.jedis.JedisPoolConfig;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.InputStream;
import java.util.Map;
import java.util.concurrent.TimeUnit;

public class RedisUtil {
    private static JedisPool jedisPool;
    private static String HOST;
    private static int PORT;
    private static String PASSWORD;
    private static int DATABASE;
    private static int MAX_ACTIVE;
    private static int MAX_IDLE;
    private static int MIN_IDLE;
    private static int MAX_WAIT;

    static {
        loadConfig();
        initPool();
    }

    private static void loadConfig() {
        try {
            Yaml yaml = new Yaml();
            InputStream inputStream = new FileInputStream("redis.page");
            Map<String, Object> config = yaml.load(inputStream);

            Map<String, Object> spring = (Map<String, Object>) config.get("spring");
            Map<String, Object> redis = (Map<String, Object>) spring.get("redis");

            HOST = (String) redis.getOrDefault("host", "127.0.0.1");
            PORT = (Integer) redis.getOrDefault("port", 6379);
            PASSWORD = (String) redis.getOrDefault("password", "");
            DATABASE = (Integer) redis.getOrDefault("database", 0);

            Map<String, Object> lettuce = (Map<String, Object>) redis.get("lettuce");
            Map<String, Object> pool = (Map<String, Object>) lettuce.get("pool");

            MAX_ACTIVE = (Integer) pool.getOrDefault("max-active", 200);
            MAX_IDLE = (Integer) pool.getOrDefault("max-idle", 50);
            MIN_IDLE = (Integer) pool.getOrDefault("min-idle", 10);
            MAX_WAIT = parseTime((String) pool.getOrDefault("max-wait", "100ms"));

            inputStream.close();
        } catch (FileNotFoundException e) {
            HOST = "127.0.0.1";
            PORT = 6379;
            PASSWORD = "";
            DATABASE = 0;
            MAX_ACTIVE = 200;
            MAX_IDLE = 50;
            MIN_IDLE = 10;
            MAX_WAIT = 100;
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private static int parseTime(String time) {
        if (time.endsWith("ms")) {
            return Integer.parseInt(time.replace("ms", ""));
        }
        return Integer.parseInt(time);
    }

    private static void initPool() {
        JedisPoolConfig config = new JedisPoolConfig();
        config.setMaxTotal(MAX_ACTIVE);
        config.setMaxIdle(MAX_IDLE);
        config.setMinIdle(MIN_IDLE);
        config.setMaxWaitMillis(MAX_WAIT);
        config.setTestOnBorrow(true);
        config.setTestOnReturn(false);
        config.setTestWhileIdle(true);

        jedisPool = new JedisPool(config, HOST, PORT, 2000,
                PASSWORD == null || PASSWORD.isEmpty() ? null : PASSWORD, DATABASE);
    }

    public static Jedis getConnection() {
        return jedisPool.getResource();
    }

    public static String get(String key) {
        try (Jedis jedis = getConnection()) {
            return jedis.get(key);
        }
    }

    public static String set(String key, String value) {
        try (Jedis jedis = getConnection()) {
            return jedis.set(key, value);
        }
    }

    public static String set(String key, String value, long timeout, TimeUnit unit) {
        try (Jedis jedis = getConnection()) {
            return jedis.set(key, value, "NX", "EX", unit.toSeconds(timeout));
        }
    }

    public static Long del(String key) {
        try (Jedis jedis = getConnection()) {
            return jedis.del(key);
        }
    }

    public static Boolean exists(String key) {
        try (Jedis jedis = getConnection()) {
            return jedis.exists(key);
        }
    }

    public static Long incr(String key) {
        try (Jedis jedis = getConnection()) {
            return jedis.incr(key);
        }
    }

    public static Long decr(String key) {
        try (Jedis jedis = getConnection()) {
            return jedis.decr(key);
        }
    }

    public static void close() {
        if (jedisPool != null && !jedisPool.isClosed()) {
            jedisPool.close();
        }
    }
}

File: utils/redis_util.py
import yaml
import redis
from redis.connection import ConnectionPool

class RedisUtil:
    _pool = None
    _config = None

    @classmethod
    def _load_config(cls):
        if cls._config is not None:
            return cls._config

        try:
            with open('redis.page', 'r', encoding='utf-8') as f:
                cls._config = yaml.safe_load(f)
        except FileNotFoundError:
            cls._config = {
                'spring': {
                    'redis': {
                        'host': '127.0.0.1',
                        'port': 6379,
                        'password': '',
                        'database': 0,
                        'lettuce': {
                            'pool': {
                                'max-active': 200,
                                'max-idle': 50,
                                'min-idle': 10,
                                'max-wait': '100ms'
                            }
                        }
                    }
                }
            }
        return cls._config

    @classmethod
    def _get_pool(cls):
        if cls._pool is not None:
            return cls._pool

        config = cls._load_config()
        redis_config = config['spring']['redis']
        pool_config = redis_config['lettuce']['pool']

        max_wait = pool_config.get('max-wait', '100ms')
        if isinstance(max_wait, str) and max_wait.endswith('ms'):
            max_wait_ms = int(max_wait.replace('ms', ''))
        else:
            max_wait_ms = int(max_wait)

        cls._pool = ConnectionPool(
            host=redis_config.get('host', '127.0.0.1'),
            port=redis_config.get('port', 6379),
            password=redis_config.get('password') or None,
            db=redis_config.get('database', 0),
            max_connections=pool_config.get('max-active', 200),
            max_idle=int(pool_config.get('max-idle', 50)),
            min_idle=int(pool_config.get('min-idle', 10)),
            socket_timeout=max_wait_ms / 1000,
            socket_connect_timeout=max_wait_ms / 1000
        )
        return cls._pool

    @classmethod
    def get_connection(cls):
        return redis.Redis(connection_pool=cls._get_pool())

    @classmethod
    def get(cls, key):
        with cls.get_connection() as client:
            return client.get(key)

    @classmethod
    def set(cls, key, value, ex=None, px=None, nx=False, xx=False):
        with cls.get_connection() as client:
            return client.set(key, value, ex=ex, px=px, nx=nx, xx=xx)

    @classmethod
    def setnx(cls, key, value):
        with cls.get_connection() as client:
            return client.setnx(key, value)

    @classmethod
    def setex(cls, key, time, value):
        with cls.get_connection() as client:
            return client.setex(key, time, value)

    @classmethod
    def delete(cls, *keys):
        with cls.get_connection() as client:
            return client.delete(*keys)

    @classmethod
    def exists(cls, *keys):
        with cls.get_connection() as client:
            return client.exists(*keys)

    @classmethod
    def incr(cls, key):
        with cls.get_connection() as client:
            return client.incr(key)

    @classmethod
    def decr(cls, key):
        with cls.get_connection() as client:
            return client.decr(key)

    @classmethod
    def hget(cls, name, key):
        with cls.get_connection() as client:
            return client.hget(name, key)

    @classmethod
    def hset(cls, name, key, value):
        with cls.get_connection() as client:
            return client.hset(name, key, value)

    @classmethod
    def hgetall(cls, name):
        with cls.get_connection() as client:
            return client.hgetall(name)

    @classmethod
    def lpush(cls, name, *values):
        with cls.get_connection() as client:
            return client.lpush(name, *values)

    @classmethod
    def rpop(cls, name):
        with cls.get_connection() as client:
            return client.rpop(name)

    @classmethod
    def sadd(cls, name, *values):
        with cls.get_connection() as client:
            return client.sadd(name, *values)

    @classmethod
    def smembers(cls, name):
        with cls.get_connection() as client:
            return client.smembers(name)

    @classmethod
    def expire(cls, name, time):
        with cls.get_connection() as client:
            return client.expire(name, time)

    @classmethod
    def ttl(cls, name):
        with cls.get_connection() as client:
            return client.ttl(name)

    @classmethod
    def close(cls):
        if cls._pool is not None:
            cls._pool.disconnect()
            cls._pool = None